### PR TITLE
feat: add k8s.provider.id to all traces and logs sent by the operator

### PR DESCRIPTION
**Pull request by mmanciop**
Kubernetes has no notion of cluster identity. EKS clusters, however, use the EC2 instance identifier inside the `v1/Node.spec.providerID` field. This PR adds to all telemetry forwarded by the `telemetry-proxy` a new `k8s.provider.id` resource attribute, that we can use in the Lumigo backend to correlate the telemetry, and the EKS cluster that sent it.

### CODE REVIEW

**Comment by mmanciop** [comment]
TODO: Implement kind and EKS tests

File: tests/kubernetes-distros/kind/lumigooperator_traces_test.go [hunk 1]
```diff
@@ -421,6 +422,49 @@ func TestLumigoOperatorTraces(t *testing.T) {
 
 			return ctx
 		}).
+		Assess("All traces have the 'k8s.provider.id' taken from describe node", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
```
Comment (mmanciop) [inline tests/kubernetes-distros/kind/lumigooperator_traces_test.go:425]: Change it to: "All traces have the 'k8s.provider.id' resource attribute set"

File: tests/kubernetes-distros/kind/lumigooperator_traces_test.go [hunk 2]
```diff
@@ -421,6 +422,49 @@ func TestLumigoOperatorTraces(t *testing.T) {
...
+				for i := 0; i < l; i++ {
+					resourceSpans := exportRequest.Traces().ResourceSpans().At(i)
+
+					resourceAttributes := resourceSpans.Resource().Attributes().AsRaw()
+
+					if actualClusterUID, found := resourceAttributes["k8s.provider.id"]; !found {
+						t.Fatalf("found spans without the 'k8s.provider.id' resource attribute: %+v", resourceAttributes)
+					} else {
+						actualClusterUIDstr, ok := actualClusterUID.(string)
+						if !ok || !strings.HasPrefix(actualClusterUIDstr, "kind://") {
```
Comment (mmanciop) [inline tests/kubernetes-distros/kind/lumigooperator_traces_test.go:459]: This test is not strong enough IMO. You actually could add the cluster name as a context param and read it in the test

**Review by mmanciop — COMMENTED** [review]
@moshe-shaham-lumigo thjere should be one "cdk*" folder, not two :-)

**Review by moshe-shaham-lumigo — APPROVED** [review]

**Pull request merged**